src/arrayMethodReduce.js

```diff
@@ -5,7 +5,38 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    if (typeof callback !== 'function') {
+      throw new Error('not a function');
+    }
+
+    if (this.length === 0 && !initialValue) {
+      throw new Error('no initial value and array is empty');
+    }
+
+    let k = 0;
+    let accumulator = typeof this[0] === 'number'
+      ? 0
+      : '';
+    let kPresent;
+
+    if (arguments.length === 2) {
+      accumulator = initialValue;
```

Comment:
I think checking that we don't have enough arguments would be more appropriate because we expect to get at least 2 arguments for our function to work and if we have fewer, then we should do some actions. IMHO